src/lib/sql/sql.hpp

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "types.hpp"
+
+#include "sql_pipeline.hpp"
+#include "sql_pipeline_statement.hpp"
+
+namespace opossum {
+
+class Optimizer;
+
+/**
+ * Interface for the configured execution of SQL.
+ *
+ * Minimal usage:
+ *      SQL{"SELECT * FROM t;"}.pipeline().get_result_table()
+ *
+ * With custom Optimizer and TransactionContext:
+ *      SQL{query}.
+ *          set_optimizer(optimizer).
+ *          set_transaction_context(tc).
+ *          pipeline();
+ *
+ * Defaults:
+ *  - MVCC is enabled
+ *  - The default Optimizer (Optimizer::create_default_optimizer() is used.
+ *
+ * Favour this interface over calling the SQLPipeline[Statement] constructors with their long parameter list.
```

Comment:
Do we want this to be the one-and-only way to execute SQL? Maybe the `SQLPipeline` constructor should be protected and this class friended?

Reply:
I'm not the biggest fan of forbidding object creation to enforce preferred usage patterns. Will not fight over it in this case though. Your call.

Reply:
Nope, just an idea